activesupport/lib/active_support/deprecation/behaviors.rb

```diff
@@ -114,7 +114,7 @@ def arity_coerce(behavior)
             raise ArgumentError, "#{behavior.inspect} is not a valid deprecation behavior."
           end
 
-          if behavior.arity == 4 || behavior.arity == -1
+          if !behavior.respond_to?(:arity) || behavior.arity == 4 || behavior.arity == -1
             behavior
           else
             -> message, callstack, _, _ { behavior.call(message, callstack) }
```

Comment:
Since this `if` / `else` is to preserve backward compatibility of old `behavior` values (see #28800), it might be better to flip it around:

 ```suggestion
          if behavior.respond_to?(:arity) && behavior.arity == 2
            -> message, callstack, _, _ { behavior.call(message, callstack) }
          else
            behavior
```

That way, if `behavior.arity` is 0 or 1 or 5+, the "wrong number of arguments" error will specify "given 4" instead of "given 2".